### ISSUE TITLE
Update code samples with retry loops for connection error cases

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/docs#5470]
We are currently updating our code samples with a retry loop to handle [client-side transaction retries][retries].

However, there are other error conditions we may want to handle as well.

For example, we should probably handle the error types in **Class 08 - Connection Exception** from the [PG error codes][errcodes], namely:

- [ ] `08000`: `connection_exception`
- [ ] `08003`: `connection_does_not_exist`
- [ ] `08006`: `connection_failure`
- [ ] `08001`: `sqlclient_unable_to_establish_sqlconnection`
- [ ] `08004`: `sqlserver_rejected_establishment_of_sqlconnection`
- [ ] `08007`: `transaction_resolution_unknown`
- [ ] `08P01`: `protocol_violation`

[retries]: https://www.cockroachlabs.com/docs/stable/transactions.html#client-side-intervention
[errcodes]: https://www.postgresql.org/docs/10/errcodes-appendix.html

**Comment by rmloveland** [comment]
Hey @spencerkimball ,

@jseldess asked me to create this issue based on a convo y'all had, and said you might have additional input.

**Comment by jseldess** [comment]
Closing in favor of https://github.com/cockroachdb/docs/issues/5476.

**Comment by ericharmeling** [comment]
@rmloveland 

I'm grooming the SQL backlog, and looking to triage or close issues with the `A-sql` label. Is this issue covered by #5476?

**Comment by rmloveland** [comment]
> Is this issue covered by #5476?

No, this is covering all sorts of errors that might need to be handled in your app logic that are NOT transaction retry / serialization errors.  So IMO we should not close this one.

**Comment by ericharmeling** [comment]
@rmloveland 
Gotcha. Thanks for clarifying. This sort of information could be pretty useful to developers. I've put this issue on the AppDev docs backlog for now.
